core/dbt/events/functions.py

```diff
@@ -57,7 +57,11 @@ def setup_event_logger(log_path, level_override=None):
     EVENT_HISTORY = deque(maxlen=flags.EVENT_BUFFER_SIZE)  # type: ignore
 
     make_log_dir_if_missing(log_path)
-    this.format_json = flags.LOG_FORMAT == "json"
+    # Flags is not set in the integration test suite but is by the environment.
+    # DBT_LOG_FORMAT is used in our test suite so checking for it's presence is key.
+    this.format_json = (
+        flags.LOG_FORMAT == "json" or os.environ.get("DBT_LOG_FORMAT", "json") == "json"
```

Comment:
I don't think that this is correct. If the log_format isn't set by env_var, user_config or cli option, the default is None. This is basically changing the default to be 'json'.  If the log_format isn't set and it IS set in an env_var, it means that flags.set_from_args has not been called someplace where it ought to be.

Reply:
> This is basically changing the default to be 'json'.
Yep, that was to get this working, and I think I mentioned that in Slack.
> If the log_format isn't set and it IS set in an env_var, it means that flags.set_from_args has not been called someplace where it ought to be.
See, this is exactly where I was hoping someone more familiar would know the answer. Let me see about getting that routine called and reverting this line.

Reply:
Think I've got something better now. Best part is once the base.py is removed, this patch will go with it, i.e. the fix is scoped to integration tests